src/commands/submit.ts

```diff
@@ -40,17 +40,16 @@ export default class Submit extends Command {
             return;
         }
 
-        const compiler = new AppCompiler(this, fd);
-        const report = compiler.logDiagnostics();
+        const compiler = new AppCompiler(fd);
+        const report = await compiler.compile();
 
-        if (!report.isValid) {
+        if (!report.diagnostics.length) {
             this.error('TypeScript compiler error(s) occurred');
             this.exit(1);
             return;
         }
 
-        const packager = new AppPackager(this, fd);
-        const zipName = await packager.zipItUp();
+        const zipName = await compiler.outputZip();
```

Comment:
Is this the typescript or the compiled files?

Reply:
compiled files

Reply:
oooooohhhh right